### PR TITLE
Use per target flags

### LINKED ISSUE #149: CMake Clean up
Move to per-target based flags everywhere.

### PR DESCRIPTION
**Pull request by streeve**
Closes #149 (unless I missed something)

### CODE REVIEW

**Comment by streeve** [comment]
@rfbird let me know if you have a different idea about defaults

File: core/performance_test/peakflops/CMakeLists.txt [hunk 1]
```diff
@@ -1,48 +1,44 @@
...
-set(Cabana_PERFORMANCE_VECLENGTH "" CACHE STRING "Vector Length to use when running expected flops")
-set(Cabana_PERFORMANCE_ITERATIONS "" CACHE STRING "Number of iterations to repeat the test for (increases flops)")
-set(Cabana_PERFORMANCE_SEED "" CACHE STRING "Seed for performance test RNG")
+set(Cabana_PERFORMANCE_EXPECTED_FLOPS "0" CACHE STRING "Value for expected flops in performance tests")
+set(Cabana_PERFORMANCE_ERROR_MARGIN "0" CACHE STRING "Error margin to use in performance tests")
+set(Cabana_PERFORMANCE_VECLENGTH "8" CACHE STRING "Vector Length to use when running expected flops")
+set(Cabana_PERFORMANCE_ITERATIONS "10" CACHE STRING "Number of iterations to repeat the test for (increases flops)")
+set(Cabana_PERFORMANCE_SEED "0" CACHE STRING "Seed for performance test RNG")
 
 if(NOT Cabana_PERFORMANCE_EXPECTED_FLOPS STREQUAL "")
```
Comment (junghans) [inline core/performance_test/peakflops/CMakeLists.txt:13]: If it is always define, couldn't we drop this check?

File: core/performance_test/peakflops/CMakeLists.txt [hunk 2]
```diff
@@ -1,48 +1,44 @@
...
   return()
 endif()
 
-set(Cabana_PERFORMANCE_EXPECTED_FLOPS "" CACHE STRING "Value for expected flops in performance tests")
-set(Cabana_PERFORMANCE_ERROR_MARGIN "" CACHE STRING "Error margin to use in performance tests")
-set(Cabana_PERFORMANCE_VECLENGTH "" CACHE STRING "Vector Length to use when running expected flops")
-set(Cabana_PERFORMANCE_ITERATIONS "" CACHE STRING "Number of iterations to repeat the test for (increases flops)")
-set(Cabana_PERFORMANCE_SEED "" CACHE STRING "Seed for performance test RNG")
+set(Cabana_PERFORMANCE_EXPECTED_FLOPS "0" CACHE STRING "Value for expected flops in performance tests")
+set(Cabana_PERFORMANCE_ERROR_MARGIN "0" CACHE STRING "Error margin to use in performance tests")
```
Comment (rfbird) [inline core/performance_test/peakflops/CMakeLists.txt:8]: I think the defaults should probably match what's in the code, here:

https://github.com/ECP-copa/Cabana/blob/dd7101b63411cad5208dad0299e7c417ea2a7dcf/core/performance_test/peakflops/common.h

 and 

https://github.com/ECP-copa/Cabana/blob/0f4de9cad7feadc52d1ee15e11fd0b51f7294e78/core/performance_test/peakflops/01_cpp_simple.cpp#L140

If we're hoisting the default to here, we should probably tidy those bits of the code too?

Comment (streeve) [reply core/performance_test/peakflops/CMakeLists.txt]: Much cleaner, but it does now use the same number of iterations per test

File: core/unit_test/CMakeLists.txt [hunk 3]
```diff
@@ -64,7 +62,8 @@ macro(Cabana_add_tests)
       )
       set(_target ${_test}_test_${_device})
       add_executable(${_target} ${_file} ${CABANA_UNIT_TEST_MAIN})
-      target_include_directories(${_target} PUBLIC ${_dir})
+      target_include_directories(${_target} PUBLIC ${_dir}
```
Comment (junghans) [inline core/unit_test/CMakeLists.txt:65]: It seems `PUBLIC` doesn't make sense for executables (https://reviews.llvm.org/D40823).

**Pull request merged**